unittests/test_pipeline.py

```diff
@@ -457,6 +457,37 @@ def postfoo(self):
         assert '_RFM_PRE_SETUP' in os.environ
         assert '_RFM_POST_SETUP' in os.environ
 
+    def test_setup_hooks_in_compile_only_test(self):
+        @fixtures.custom_prefix('unittests/resources/checks')
+        class MyTest(rfm.CompileOnlyRegressionTest):
+            def __init__(self):
+                self.name = 'hellocheck_compile'
+                self.descr = 'C Hello World test'
+
+                # All available systems are supported
+                self.valid_systems = ['*']
+                self.valid_prog_environs = ['*']
+                self.sourcepath = 'hello.c'
+                self.tags = {'foo', 'bar'}
+                self.name = type(self).__name__
+                self.executable = os.path.join('.', self.name)
+                self.sanity_patterns = sn.any(sn.glob(self.executable))
+
+            @rfm.run_before('setup')
+            def prefoo(self):
+                assert self.current_environ is None
+                os.environ['_RFM_PRE_SETUP_COMPILE'] = 'foo'
```

Comment:
Better increment a counter or put a value in a list, rather than modifying the environment.